scheds/rust/scx_bpfland/src/bpf/main.bpf.c

```diff
@@ -56,6 +56,13 @@ const volatile u64 slice_ns_lag;
  */
 const volatile u64 nvcsw_thresh = 10;
 
+/*
+ * Schedule up to the starvation_thresh number of priority tasks before being
+ * required to schedule a regular task. This ensures that regular tasks are not
+ * starved by interactive tasks.
+ */
+const volatile u64 starvation_thresh = 100;
```

Comment:
Down the line, it *may* lead to more consistent behavior if the mechanism is time based rather than the number of events. Another thing to consider is that if there are tasks which are affine to some subset of CPUs, they might still be able to starve, so maybe a mechanism which can transfer timing out batch tasks to per-cpu DSQs will probably be more fool-proof. Just ideas for possible future improvements.

Reply:
Yes, a time-based approach would definitely be more effective. I'll work on that.

I also like the idea of transferring tasks to per-CPU DSQs. That's something I coud use also for the CPU hotplug support, to transfer tasks to another DSQ from the `.cpu_offline()` callback, but I haven't found a way to pop tasks from a non-local DSQ and move them to another DSQ, except via `scx_bpf_consume()`, but this doesn't give me the task, it just "internally" transfers it to the local DSQ and of course I can't use consume in `cpu_offline()`...